consensus/src/main/java/org/ethereum/beacon/consensus/transition/BeaconStateExImpl.java

```diff
@@ -13,28 +13,18 @@
 @SSZSerializable(instanceGetter = "getDelegate")
 public class BeaconStateExImpl extends DelegateBeaconState implements BeaconStateEx {
 
-  private final Hash32 latestChainBlock;
   private final TransitionType lastTransition;
 
-  public BeaconStateExImpl(BeaconState canonicalState,
-      Hash32 latestChainBlock,
-      TransitionType lastTransition) {
+  public BeaconStateExImpl(BeaconState canonicalState, TransitionType lastTransition) {
     super(canonicalState);
-    this.latestChainBlock = latestChainBlock;
     this.lastTransition = lastTransition;
   }
 
   /**
    * @param canonicalState regular BeaconState
-   * @param latestChainBlock  latest block which was processed on this state chain
    */
-  public BeaconStateExImpl(BeaconState canonicalState, Hash32 latestChainBlock) {
-    this(canonicalState, latestChainBlock, TransitionType.UNKNOWN);
-  }
-
-  @Override
-  public Hash32 getHeadBlockHash() {
-    return latestChainBlock;
```

Comment:
Cool! We don't need this stuff anymore 👍 
M.b. we can think of getting rid of the `BeaconStateEx` completely?

Reply:
I was thinking about that. We can bring `BeaconStateEx` back if it will be needed later on.

Reply:
Looks like I've found at least one use case which keeps `BeaconStateEx` still valuable: https://github.com/harmony-dev/beacon-chain-java/blob/develop/validator/src/main/java/org/ethereum/beacon/validator/MultiValidatorService.java#L220